Enterprise-CMCS/managed-care-review
# MCR-4023, MCR-3975 delete old rate tables + submissionPackages on fetchRate

## Summary

This PR completes a majority of our cleanup work coming out of the linked rates work. 

This PR also adds `packageSubmissions` to the rate type, and adjusts rate and contract submit to be based on the same code filling out new package submissions for both. 


Example tasks:
- [x] remove outdated tables from Prisma https://github.com/Enterprise-CMCS/managed-care-review/pull/2517/commits/64cb494430805671113f6a17c7e0058e77419fd3
- [x] update ContractType domain model for draft revisions so that  draft contract revisions, find their related rates following the  top level `draftRates` https://github.com/Enterprise-CMCS/managed-care-review/pull/2517/commits/78d941c3102ff8eeb4a8813995608f40fedec9f9
- [x] remove prisma helpers for drafts https://github.com/Enterprise-CMCS/managed-care-review/pull/2517/commits/78d941c3102ff8eeb4a8813995608f40fedec9f9
- [x] do not return an `Error` for Rate parse anymore (match behavior from Contract) https://github.com/Enterprise-CMCS/managed-care-review/pull/2517/commits/2ceda13916b753e2a2ce45d647b82121f6f23939
- [x]  fix `submitRate` – get tests passing [2ed4062](https://github.com/Enterprise-CMCS/managed-care-review/pull/2517/commits/2ed4062812302f6847d82b2afeb56c369bbdbb85)
- [x] fix `submitContract` – get tests passing [61b88d2](https://github.com/Enterprise-CMCS/managed-care-review/pull/2517/commits/61b88d2c1e2a8b33d9325f49600ec7bd8751840e)
- [x] update RateType domain model for draft revisions
    - specifically this PR addresses the types infinity loop issue that has been problematic for us - adds new `baseContractType` and `baseRateType` to zod
- [x] add new `RatePackageSubmissions` type to graphql
- [x] add packageSubmissions to fetchRate [1628ef2](https://github.com/Enterprise-CMCS/managed-care-review/pull/2517/commits/1628ef2408b6805b9ff51b6fa67aaa7c9a147de8)
---

#### Related issues
https://jiraent.cms.gov/browse/MCR-4023
https://jiraent.cms.gov/browse/MCR-3975

#### Test cases covered
`rate returns packageSubmissions` 

## QA guidance
Check for regressions around submit (for child rates, linked rates, combination, unlock, resubmit) in the review app.
